frontend/src/component/Activity/Preview.jsx

```diff
@@ -0,0 +1,222 @@
+import React from 'react'
+import { translate } from 'react-i18next'
+import { Link } from 'react-router-dom'
+import {
+  buildFilePreviewUrl as jpgPreviewUrl,
+  removeExtensionOfFilename,
+  removeInteractiveContentFromHTML,
+  SCREEN_SIZE
+} from 'tracim_frontend_lib'
+import { FETCH_CONFIG, PAGE } from '../../util/helper.js'
+import PropTypes from 'prop-types'
+import classnames from 'classnames'
+
+require('./Preview.styl')
+
+// possible preview sizes with their media queries, ordered from smallest to biggest
+const PREVIEW_WIDTHS = [
+  [
+    `(max-width: ${SCREEN_SIZE.MAX_SM})`,
+    376 // 400px - margins (10 * 2px) - borders (2 * 1px)
+  ],
+  [
+    '',
+    776 // 800px - margins (10 * 2px) - borders (2 * 1px)
+  ]
+]
+
+const MAX_PREVIEW_HEIGHT = 300
+
+class Preview extends React.Component {
+  state = {
+    preview: this.isHtmlPreview() ? null : this.getJPEGPreview(),
+    previewOverflow: false,
+    previewLoading: this.isHtmlPreview(),
+    previewUnavailable: false
+  }
```

Comment:
Usually the initial state is set inside the constructor. Any reason for not doing this?

Reply:
I found this syntax while looking for help today.  I find it cleaner, clearer, simpler and more concise than putting it in a constructor and having to call super(props).

I'll ask the others if they are fine with this.